passw0rd/src/main/kotlin/com/virgilsecurity/passw0rd/protocol/Protocol.kt

```diff
@@ -119,53 +119,50 @@ class Protocol @JvmOverloads constructor(
      * @throws InvalidPasswordException
      * @throws InvalidProtobufTypeException
      */
-    @Throws(IllegalArgumentException::class,
+    @Throws(
+            IllegalArgumentException::class,
             ProtocolException::class,
             PheException::class,
             InvalidPasswordException::class,
-            InvalidProtobufTypeException::class)
+            InvalidProtobufTypeException::class
+    )
     fun verifyPassword(password: String, enrollmentRecord: ByteArray): CompletableFuture<ByteArray> = GlobalScope.async {
-        if (password.isBlank()) Utils.shouldNotBeEmpty("password")
-        if (enrollmentRecord.isEmpty()) Utils.shouldNotBeEmpty("enrollmentRecord")
+        requires(password.isNotBlank(), "password")
+        requires(enrollmentRecord.isNotEmpty(), "enrollmentRecord")
 
-        val (version, record) = try {
-            Passw0rdProtos.DatabaseRecord.parseFrom(enrollmentRecord).let {
-                it.version to it.record.toByteArray()
-            }
+        val databaseRecord = try {
```

Comment:
The same question - why don't use let function and make it more concise?

Reply:
Same as above :)